src/ResourceManager/Profile/Commands.Profile/AzureRmAlias/DisableAzureRmAlias.cs

```diff
@@ -0,0 +1,188 @@
+                    }
+                }
+            }
+
+            // Set path to user profile based on the Scope
+            object userprofile = "";
+            if (Scope != null && Scope.Equals("CurrentUser"))
+            {
+                userprofile = SessionState.PSVariable.GetValue("env:USERPROFILE") + "\\Documents\\PowerShell\\profile.ps1";
+            }
+
+            else if (Scope != null && Scope.Equals("LocalMachine"))
+            {
+                userprofile = SessionState.PSVariable.GetValue("PSHOME") + "\\profile.ps1";
+            }
+
+            if (Scope != null && (Scope.Equals("CurrentUser") || Scope.Equals("LocalMachine")))
+            {
+                if (File.Exists(userprofile.ToString()))
+                {
+                    List<string> modulesToKeep = new List<String>();
+                    string filecontent = "";
+                    string originalText = File.ReadAllText(userprofile.ToString());
+                    if (originalText.Contains("#Begin Azure PowerShell alias import"))
+                    {
+                        // Add back profile code unrelated to Azure PowerShell aliases
+                        var splitOriginalText = originalText.Split(new string[] { "#Begin Azure PowerShell alias import", "#End Azure PowerShell alias import" }, StringSplitOptions.None);
+                        filecontent += splitOriginalText[0];
+                        if (splitOriginalText.Length > 2)
+                        {
+                            filecontent += splitOriginalText[2];
+                        }
+
+                        if (Module != null)
+                        {
+                            // Create list of modules to keep (all modules currently enabled in the profile not listed in -Module)
+                            var regex = new Regex(@"Az\.[a-zA-Z0-9\.]+(,\s|\s-)");
+                            Match match = regex.Match(splitOriginalText[1].Split(new string[] { "Import-Module Az.Profile" }, StringSplitOptions.None)[1]);
+                            while (match.Success)
+                            {
+                                if (!Module.Contains(match.ToString().Substring(0, match.ToString().Length - 2), StringComparer.CurrentCultureIgnoreCase))
+                                {
+                                    modulesToKeep.Add(match.ToString().Substring(0, match.ToString().Length - 2));
+                                }
+                                match = match.NextMatch();
+                            }
+                        }
+                    }
+
+                    // Add script to enable aliases to profile if there are any modules to keep
+                    if (modulesToKeep.Count > 0)
+                    {
+                        filecontent = "#Begin Azure PowerShell alias import" + Environment.NewLine + "`$error.clear()" + Environment.NewLine +
```

Comment:
It seems like we should factor out the manipulation of the file lists into a helper class, as a lot of this should be shared with the other cmdlet